Разметка всех страниц

Alex88GitHub opened the pull request:
Разметил index.html и catalog.html

---
:mortar_board: [Разметка всех страниц](https://up.htmlacademy.ru/htmlcss/36/user/2254559/tasks/5)

:boom: https://htmlacademy-htmlcss.github.io/2254559-sedona-36/2/

Alex88GitHub committed:
Добавляет catalog

Alex88GitHub committed:
Добавляет ссылку на наставника

Alex88GitHub committed:
Merge branch 'htmlacademy-htmlcss:master' into master

Alex88GitHub committed:
Update index.html

Корректирует section "hero"

Alex88GitHub committed:
Update catalog.html

Удаляет header в main и корректирует section и "хлебные крошки".

Alex88GitHub committed:
Update catalog.html

Добавляет поля со стоимостью в фильтр

Alex88GitHub committed:
Update catalog.html

Корректирует описание ссылок "Вид представления"

Alex88GitHub committed:
Update catalog.html

Добавляет текст  результатов поиска

Alex88GitHub committed:
Update catalog.html

Корректирует результат поиска

Alex88GitHub committed:
Update catalog.html

Добавляет карточки отелей

Alex88GitHub committed:
Update catalog.html

Добавляет alt изображениям в карточках отелей

Alex88GitHub committed:
Update catalog.html

Добавляет разметку звезд в карточках отелей

Alex88GitHub committed:
Update catalog.html

Добавляет разметку рейтинга в карточки отелей

4elbik commented on index.html:
```diff
@@ -1,12 +1,148 @@
+      <p class="reasons">Рассмотрим причины, по которым Седона круче, чем Гранд-Каньон!</p>
+    </section>
+    <section class="advantages">
+      <h2 class="advantages-title vissually-hidden">Преимущества</h2>
```
Класс `advantages-title` избыточен. Мы все равно добавляем к этому тегу `visually-hidden`, который его доступно скроет, а значит и каких-либо дополнительных стилей к этому элементу нам писать и не потребуется. 
Давай его уберем, оставив только `visually-hidden`

Alex88GitHub replied:
Логично, поправил)

4elbik commented on index.html:
```diff
@@ -1,12 +1,148 @@
+        <li class="advantages-item">
+          <h3 class="advantages-item-title">Настоящий городок</h3>
+          <p class="advantages-discription">Седона — не аттракцион для туристов, там течёт своя жизнь</p>
+          <img class="advantages-photo-city" src="#" alt="фотография города">
```
1. Вот у нас есть заголовок с классом `advantages-item-title`. Здесь понятно, что у нас это заголовок конкретного элемента преимуществ.
А в классе `advantages-discription` у нас как будто описание для всех преимуществ. Предлагаю добавить уточнение, что это конкретно элемент преимуществ

2. То же относиться и к картинке, но с небольшим дополнением. Нам нужно добавить и одновременно убрать уточнение)) Что я имею ввиду - `advantages-item-photo` названия будет вполне достаточно. (добавили уточнение (item), что это конкретно к элементу преимуществ и убрали уточнение (city), что там внутри этого элемента за картинка)

Alex88GitHub replied:
Понятно, поправил. По п.2 получается, что наш код становиться универсальным, т.е. классы не привязываются к содержанию фото. В дальнейшем можно менять контент (фото), без изменения кода.

4elbik commented on index.html:
```diff
@@ -1,12 +1,148 @@
+      <h2 class="advantages-service-title">Приезжайте в Седону отдохнуть в комфорте и уюте!</h2>
+      <p class="advantages-service-description">Опытный персонал и качественное обслуживание!</p>
+      <ul class="advantages-service-list">
+        <li class="advantages-service-item house">
```
Отличное добавление ещё одного класса, для уточнения, что там у нас внутри будет за картинка 👍

4elbik commented on index.html:
```diff
@@ -1,12 +1,148 @@
+    </section>
+    <section class="search-hotels">
+      <h2 class="visually-hidden">Поиск отеля</h2>
+      <h3 class="">Заинтересовались?</h3>
```
Кажется, здесь пропал класс)

Alex88GitHub replied:
Добавил.

4elbik commented on index.html:
```diff
@@ -1,12 +1,148 @@
+      <h3 class="">Заинтересовались?</h3>
+      <p class="search-hotels-description">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения
+        гостиниц в Седоне</p>
+      <button class="button" type="button">ПОИСК ГОСТИНИЦЫ В СЕДОНЕ</button>
```
КАПСЛОК В ИСХОДНИКАХ - ТАКОЕ СЕБЕ ДЕЛО))

Alex88GitHub replied:
Согласен, не очень смотрится. Исправил.

4elbik commented on index.html:
```diff
@@ -1,12 +1,148 @@
+      <p class="subscription-description">Только полезная информация и никакого спама, честное бойскаутское!</p>
+      <form class="subscription-form" action="https://echo.htmlacademy.ru/" method="post">
+        <input type="email" name="email" placeholder="Ваш e-mail" required>
+        <button class=" button button-subscription" type="submit">Подписаться</button>
```
Лишний пробел перед ` button`

Alex88GitHub replied:
Сделал

4elbik commented on index.html:
```diff
@@ -1,12 +1,148 @@
+      <button class="button" type="button">ПОИСК ГОСТИНИЦЫ В СЕДОНЕ</button>
+    </section>
+    <section class="subscription">
+      <h2 class="subscription-title">Подпишитесь на рассылку</h2>
```
Эта секция у нас повторяется и на главной странице, и на внутренней. В **нижней** части страницы

Alex88GitHub replied:
Ты намекаешь, что эту секцию надо определить в footer?)

4elbik commented on catalog.html:
```diff
@@ -0,0 +1,235 @@
+    <nav class="navigation">
+      <ul class="navigation-menu">
+        <li class="navigation-menu-item">
+          <a class="navigation-link navigation-link-current" href="index.html">Главная</a>
```
На главной `href` нет, здесь он есть. Просто идеально!))
Это на самом деле очень частая ошибка на защите, и очень круто, что ты это предусмотрел с самой первой домашки!))

4elbik commented on catalog.html:
```diff
@@ -0,0 +1,235 @@
+      <h2 class="visually-hidden">Поиск с фильтром</h2>
+      <ol class="breadcrumbs">
+        <li class="breadcrumbs-item">
+          <a class="main-link-image" href="index.html"></a>
```
Стоит докинуть внутрь текст для доступности. Куда мы перейдем при клике на эту ссылку

Alex88GitHub replied:
Подразумевается в дальнейшем использование декоративной иконки в виде домика. Если добавим внутрь &lt;a&gt; "Главная", то потом как это подпись с картинкой себя поведет?

4elbik commented on catalog.html:
```diff
@@ -0,0 +1,235 @@
+      </ol>
+      <form class="catalog-filter" action="https://echo.htmlacademy.ru/" method="get">
+        <button class="button" type="submit">Применить</button>
+        <button type="reset">Сбросить</button>
```
Давай баттоны перекинем в самый конец формы?
Если сейчас идти по сайту через фокусирование (с помощью таба), то несовсем логичное поведение будет ожидаться, когда мы сразу фокусируемся на кнопках отправки/обнуления формы, которую мы вообще ещё даже и не трогали/заполняли

Alex88GitHub replied:
сделал

4elbik commented on catalog.html:
```diff
@@ -0,0 +1,235 @@
+          </ul>
+        </fieldset>
+        <fieldset class="catalog-filter-group">
+          <legend class="catalog-filter-type-houses">Тип жилья:</legend>
```
`catalog-filter-type-houses` -> `catalog-filter-title`

Зачем нам здесь уточнение в классе, что это "тип жилья"? Визуально "инфраструктура" и "тип жилья" одинаково застилизованы по макету. Так тогда давай у них сделаем одинаковый класс

Alex88GitHub replied:
Согласен, тогда логично и у блока стоимости применить тот же класс catalog-filter-title

Alex88GitHub replied:
Исправил

4elbik commented on catalog.html:
```diff
@@ -0,0 +1,235 @@
+          </ul>
+        </fieldset>
+        <fieldset class="catalog-filter-group">
+          <legend class="catalog-filter-price">Стоимость,Р:</legend>
```
То же самое

Alex88GitHub replied:
Исправил

4elbik commented on catalog.html:
```diff
@@ -0,0 +1,235 @@
+    <section class="catalog-search-results">
+      <h2 class="visually-hidden">Результаты поиска</h2>
+      <p class="search-results-text">Найдено гостиниц:
+        <span class="search-results-number">38</span>
```
Для чего у нас дополнительно обернуто "38" в `span.search-results-number`?

Alex88GitHub replied:
На лайве Николай Шабалин говорил, что можно обернуть 38 в спан, чтобы потом эта цифра автоматически менялась, после работы поискового фильтра.

4elbik commented on catalog.html:
```diff
@@ -0,0 +1,235 @@
+        <li class="house-card">
+          <img class="house-card-image" src="#" alt="Гостиница Amara Resort & Spa">
+          <h3 class="house-card-title">Amara Resort & Spa</h3>
+          <p class="description-house-card">Гостиница от 4000 р</p>
```
1. `description-house-card` -> `house-card-description`. Порядок уточнения в классе должен начинаться от родительского блока. Родитель для нас - `house-card`, значит внутренние элементы начинают название класса с `house-card-...`

2. Стоит разделить "Гостиница" и "От X р" по разным `<p>`

Alex88GitHub replied:
Исправлено.

4elbik commented on catalog.html:
```diff
@@ -0,0 +1,235 @@
+          <p class="star star-4">
+            <span class="visually-hidden">4 звезды</span>
+          </p>
+          <p class="house-card-reting">Рейтинг: 8,5</p>
```
`reting` -> `rating`

Alex88GitHub replied:
Исправил

4elbik commented on catalog.html:
```diff
@@ -0,0 +1,235 @@
+          <h3 class="house-card-title">Villas at Poco Diablo</h3>
+          <p class="description-house-card">Гостиница от 5000 р</p>
+          <а class="button button-more" href="#">Подробнее</a>
+          <button class="button button-in-favorites" type="button">В избранном</button>
```
`button-in-favorites`

За вот этот дополнительный уточняющий класс - лайк)

4elbik commented on catalog.html:
```diff
@@ -0,0 +1,235 @@
+          <a class="pagination-link pagination-disabled">1</a>
+        </li>
+        <li class="pagination-item">
+          <a class="pagination-link pagination-disabled" href="#">2</a>
```
2, 3, 4, 10... Эти страницы разве задизейблены?

Alex88GitHub replied:
Конечно нет, невнимательность после ctrl+v

Alex88GitHub committed:
Update catalog.html

удаляет некорректный disabled в pagination

Alex88GitHub committed:
Update catalog.html

Исправляет на rAting

Alex88GitHub committed:
Update catalog.html

Исправляет описание класса на house-card-description. Разделяет на 2 тега<p> описание карточки.

Alex88GitHub committed:
Update index.html

Исправляет капслок

Alex88GitHub committed:
Update index.html

Добавляет название класса

Alex88GitHub committed:
Update index.html

Корректирует описание классов преимуществ.

Alex88GitHub committed:
Update index.html

удаляет класс

Alex88GitHub committed:
Update catalog.html

Исправляет тег ("a" был напечатан на рус. раскладке)

Alex88GitHub committed:
Update index.html

убирает лишний пробел

Alex88GitHub committed:
Update catalog.html

убирает пробел

Alex88GitHub committed:
Update catalog.html

Переносит button в конец формы

Alex88GitHub committed:
Update catalog.html

прописывает один класс для подписей полей фильтров

Alex88GitHub committed:
Update catalog.html

добавляет описание ссылки

Alex88GitHub committed:
Разметка страницы

Перемещает подписку в footer
